### ISSUE TITLE
Build failures with Cython 3.0.0b1

### ISSUE THREAD

**Issue by mgorny** [PyTables/PyTables#1003]
Cython 3.0.0b1 has been released recently, and pytables fails to build with it:

```pytb
$ python -m build -w
* Creating virtualenv isolated environment...
* Installing packages in isolated environment... (Cython >=3.0.0b1, blosc2 ~=2.0.0, oldest-supported-numpy, packaging, py-cpuinfo, setuptools >=61.0.0, wheel)
* Getting build dependencies for wheel...
* Using Python 3.11.2 (main, Feb 15 2023, 08:19:40) [GCC 12.2.1 20230121]
* Found cython 3.0.0b1
* USE_PKGCONFIG: True
* Found HDF5 headers at ``/usr/include``, library at ``/usr/lib64``.
* Found LZO 2 headers at ``/usr/include``, library at ``/usr/lib64``.
.. WARNING:: Could not find the LZO 2 runtime.
   The LZO 2 shared library was *not* found in the default library
   paths. In case of runtime problems, please remember to install it.
* Skipping detection of LZO 1 since LZO 2 has already been found.
* Found bzip2 headers at ``/usr/include``, library at ``/usr/lib64``.
.. WARNING:: Could not find the bzip2 runtime.
   The bzip2 shared library was *not* found in the default library
   paths. In case of runtime problems, please remember to install it.
* Found blosc headers at ``/usr/include``, library at ``/usr/lib64``.
* Found blosc2 headers at ``/tmp/build-env-c08sfzmk/include``, library at ``/tmp/build-env-c08sfzmk/lib64``.
Compiling tables/utilsextension.pyx because it changed.
[1/1] Cythonizing tables/utilsextension.pyx

Error compiling Cython file:
------------------------------------------------------------
...


def _dump_h5_backtrace():
    cdef object bt = []

    if H5Ewalk(H5E_DEFAULT, H5E_WALK_DOWNWARD, e_walk_cb, <void*>bt) < 0:
                                               ^
------------------------------------------------------------

tables/utilsextension.pyx:384:47: Cannot assign type 'herr_t (unsigned int, const H5E_error_t *, void *) except? -1 nogil' to 'H5E_walk_t'
Traceback (most recent call last):
  File "/usr/lib/python3.11/site-packages/pyproject_hooks/_in_process/_in_process.py", line 353, in <module>
    main()
  File "/usr/lib/python3.11/site-packages/pyproject_hooks/_in_process/_in_process.py", line 335, in main
    json_out['return_val'] = hook(**hook_input['kwargs'])
                             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/pyproject_hooks/_in_process/_in_process.py", line 118, in get_requires_for_build_wheel
    return hook(config_settings)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/tmp/build-env-c08sfzmk/lib/python3.11/site-packages/setuptools/build_meta.py", line 338, in get_requires_for_build_wheel
    return self._get_build_requires(config_settings, requirements=['wheel'])
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/tmp/build-env-c08sfzmk/lib/python3.11/site-packages/setuptools/build_meta.py", line 320, in _get_build_requires
    self.run_setup()
  File "/tmp/build-env-c08sfzmk/lib/python3.11/site-packages/setuptools/build_meta.py", line 335, in run_setup
    exec(code, locals())
  File "<string>", line 926, in <module>
  File "<string>", line 921, in get_cython_extfiles
  File "/tmp/build-env-c08sfzmk/lib/python3.11/site-packages/Cython/Build/Dependencies.py", line 1134, in cythonize
    cythonize_one(*args)
  File "/tmp/build-env-c08sfzmk/lib/python3.11/site-packages/Cython/Build/Dependencies.py", line 1301, in cythonize_one
    raise CompileError(None, pyx_file)
Cython.Compiler.Errors.CompileError: tables/utilsextension.pyx

ERROR Backend subprocess exited when trying to invoke get_requires_for_build_wheel
```

To reproduce this, you can patch `pyproject.toml` as follows:

```diff
diff --git a/pyproject.toml b/pyproject.toml
index a6de0f6c..7992c761 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -5,7 +5,7 @@ requires = [
     "oldest-supported-numpy",
     "packaging",
     "py-cpuinfo",
-    "Cython >=0.29.21",
+    "Cython >=3.0.0b1",
     "blosc2 ~=2.0.0",
 ]
 build-backend = "setuptools.build_meta"
```

**Comment by avalentino** [comment]
@mgorny thanks for reporting.
Will try to give it a look ASAP.

**Comment by mgorny** [comment]
Thanks. It's not super urgent but it would be nice to have a "fixed" release out before 3.0.0 final is released (pip doesn't install betas by default).
